app.rb

```diff
@@ -91,18 +123,92 @@ def list_people
     end
   end
 
-  def list_rentals
-    if @rentals.empty?
-      puts 'There are no rentals to show'
-    else
-      puts 'ID of person: '
-      person_id = gets.chomp.to_i
-      puts 'Rentals: '
-      @rentals.each do |rental|
-        if person_id == rental.person.id
-          puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
-        end
+  # def list_rentals
+  #   if @rentals.empty?
+  #     puts 'There are no rentals to show'
+  #   else
+  #     puts 'ID of person: '
+  #     person_id = gets.chomp.to_i
+  #     puts 'Rentals: '
+  #     @rentals.each do |rental|
+  #       if person_id == rental.person.id
+  #         puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+  #       end
+  def load_data
+    load_books
+    load_people
+  end
+
+  def load_books
+    @books = ReadFile.new('books.json').read.map { |book| Book.new(book['title'], book['author']) }
+  end
+
+  def load_people
+    people_data = ReadFile.new('people.json').read || []
+    students = []
+    teachers = []
+
+    people_data.each do |person|
+      if person['type'] == 'student'
+        students.push(load_student(person))
+      elsif person['type'] == 'teacher'
+        teachers.push(load_teacher(person))
       end
     end
+    @people = students + teachers
+  end
+
+  def load_student(data)
+    age = data['age']
+    name = data.key?('name') ? data['name'] : 'Unknown'
+    student = Student.new(age, data['parent_permission'], name: name)
+    student.id = data['id']
+    student
+  end
+
+  def load_teacher(data)
+    age = data['age']
+    specialization = data['specialization']
+    name = data.key?('name') ? data['name'] : 'Unknown'
+    teacher = Teacher.new(age, specialization, name: name)
+    teacher.id = data['id']
+    teacher
+  end
+
+  def save_data
+    save_books
+    save_people
+    save_rentals
+  end
+
+  def save_books
+    books_data = @books.map { |book| { title: book.title, author: book.author } }
+    WriteFile.new('books.json').write(books_data)
+  end
+
+  def save_rentals
+    return unless @rentals.any?
+
+    ReadFile.new('rentals.json').read || ([] + @rentals.map do |rental|
+      {
+        date: rental.date,
+        book: { title: rental.book.title, author: rental.book.author },
+        person: { type: rental.person.class.to_s, id: rental.person.id, age: rental.person.age,
+                  name: rental.person.name }
+      }
+    end)
+    WriteFile.new('rentals.json').write(rentals_data)
```

Comment:
- [ Required ] I think it would be nice if you fix the error that occurs when writing a user's rentals to file, I encountered the error below when trying to rent a book to a user. Since I was not able to rent a book I was not able to verify that the listing of rentals works. Please see the screenshot below for more details and rectify this.

## Error Screenshot
![Screenshot from 2023-09-15 00-40-32](https://github.com/H-Ramalan/school-library/assets/16392046/895512ee-c82b-4179-876a-db08db9c8e24)